beacon_node/execution_layer/src/lib.rs

```diff
@@ -1152,29 +1160,38 @@ impl<T: EthSpec> ExecutionLayer<T> {
                     return ProvenancedPayload::try_from(relay.data.message);
                 }
 
+                let builder_boost_factor =
+                    builder_boost_factor.unwrap_or(DEFAULT_BUILDER_BOOST_FACTOR);
+
                 let relay_value = *relay.data.message.value();
+                let boosted_relay_value =
+                    relay_value.saturating_mul((builder_boost_factor / 100).into());
                 let local_value = *local.block_value();
 
-                if local_value >= relay_value {
+                if local_value >= boosted_relay_value {
                     info!(
                         self.log(),
                         "Local block is more profitable than relay block";
                         "local_block_value" => %local_value,
-                        "relay_value" => %relay_value
+                        "relay_value" => %relay_value,
+                        "boosted_relay_value" => %boosted_relay_value,
+                        "builder_boost_factor" => %builder_boost_factor,
                     );
                     return Ok(ProvenancedPayload::Local(BlockProposalContentsType::Full(
                         local.try_into()?,
                     )));
                 }
 
-                if relay_value < self.inner.builder_profit_threshold {
+                if boosted_relay_value < self.inner.builder_profit_threshold {
```

Comment:
I think this would change existing behaviour for the `builder-profit-threshold`, e.g. if user sets the threshold to 0.25 eth using the `builder-profit-threshold` flag, and it should prefer the local payload unless the relay value is greater or equal to 0.25 eth..

With this change, we'd be comparing against the boosted relay value, which means this flag won't work the same as before.